src/tribler/core/utilities/process_manager/manager.py

```diff
@@ -0,0 +1,157 @@
+from __future__ import annotations
+
+import logging
+import sqlite3
+import sys
+from pathlib import Path
+from threading import Lock
+from typing import ContextManager, List, Optional, Union
+
+from contextlib import contextmanager
+
+from tribler.core.utilities.process_manager import sql_scripts
+from tribler.core.utilities.process_manager.process import ProcessKind, TriblerProcess
+from tribler.core.utilities.process_manager.utils import with_retry
+
+logger = logging.getLogger(__name__)
+
+DB_FILENAME = 'processes.sqlite'
+
+
+global_process_manager: Optional[ProcessManager] = None
+
+_lock = Lock()
+
+
+def set_global_process_manager(process_manager: Optional[ProcessManager]):
+    global global_process_manager  # pylint: disable=global-statement
+    with _lock:
+        global_process_manager = process_manager
+
+
+def get_global_process_manager() -> Optional[ProcessManager]:
+    with _lock:
+        return global_process_manager
+
+
+def set_error(error: Union[str | Exception], replace: bool = False):
+    process_manager = get_global_process_manager()
+    if process_manager:
+        process_manager.current_process.set_error(error, replace)
+    else:
+        logger.warning('Cannot set error for process locker: no process locker global instance is set')
+
+
+class ProcessManager:
+    def __init__(self, root_dir: Path, process_kind: ProcessKind, creator_pid: Optional[int] = None,
+                 db_filename: str = DB_FILENAME):
+        self.root_dir = root_dir
+        self.db_filepath = root_dir / db_filename
+        self.connection: Optional[sqlite3.Connection] = None
+        self.current_process = TriblerProcess.current_process(self, process_kind, creator_pid)
+        self.primary_process = self.atomic_get_primary_process(self.current_process)
+
+    @property
+    def logger(self) -> logging.Logger:
+        """
+        Used by the `with_retry` decorator
+        """
+        return logger
```

Comment:
It seems more straightforward, to use a logger, initialized as a class field instead of creating a property just to pass a logger, created on a module level.

Reply:
I changed it to a field

Reply:
Agreed, I changed logger property to a simple field